components/brave_rewards/browser/extension_rewards_service_observer.cc

```diff
@@ -44,6 +44,12 @@ void ExtensionRewardsServiceObserver::OnWalletInitialized(
         extensions::api::brave_rewards::OnWalletCreated::kEventName,
         std::move(args)));
       event_router->BroadcastEvent(std::move(event));
+    } else if (result == ledger::Result::CORRUPTED_WALLET) {
```

Comment:
I know this was not introduced with this PR, but could we have only one `onWalletInitialized` and we would pass result in. This way we have only one function and handler for all of them. You could create separate issue for it and do it as follow up

Reply:
Created https://github.com/brave/brave-browser/issues/3974 to follow up

Reply:
thank you